include/libdnf/rpm/package.hpp

```diff
@@ -38,280 +38,439 @@ class Goal;
-    /// @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_nevra(DnfPackage * pkg)
+    /// @return RPM package NEVRA (Name-Epoch:Version-Release.Arch). If the Epoch is 0, it is omitted from the output.
+    /// @since 5.0
+    //
+    // @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_nevra(DnfPackage * pkg)
     std::string get_nevra() const;
 
-    /// Reurn always nevra with epoch
-    /// @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_nevra(DnfPackage * pkg)
+    /// @return RPM package NEVRA (Name-Epoch:Version-Release.Arch). The Epoch is always present even if it is 0.
+    /// @since 5.0
+    //
+    // @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_nevra(DnfPackage * pkg)
     std::string get_full_nevra() const;
 
-    /// @replaces dnf:dnf/package.py:attribute:Package.group
-    /// @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_group(DnfPackage * pkg)
+    /// @return RPM package Group (RPMTAG_GROUP)
+    /// @since 5.0
+    //
+    // @replaces dnf:dnf/package.py:attribute:Package.group
+    // @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_group(DnfPackage * pkg)
     std::string get_group() const;
 
-    /// If package is installed, return get_install_size(). Return get_download_size() otherwise.
-    /// @replaces dnf:dnf/package.py:attribute:Package.size
-    /// @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_size(DnfPackage * pkg)
+    /// @return If package is installed, return get_install_size(). Return get_package_size() otherwise.
+    /// @since 5.0
+    //
+    // TODO(dmach): drop and use get_install_size()/get_package_size() instead?
+    // @replaces dnf:dnf/package.py:attribute:Package.size
+    // @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_size(DnfPackage * pkg)
     unsigned long long get_size() const;
 
-    /// Return size of an RPM package: <size package="..."/>
-    /// @replaces dnf:dnf/package.py:attribute:Package.downloadsize
-    /// @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_downloadsize(DnfPackage * pkg)
-    unsigned long long get_download_size() const;
-
-    /// Return size of an RPM package installed on a system: <size installed="..."/>
-    /// @replaces dnf:dnf/package.py:attribute:Package.installsize
-    /// @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_installsize(DnfPackage * pkg)
+    /// @return File size of the RPM package
+    /// @since 5.0
+    //
+    // @replaces dnf:dnf/package.py:attribute:Package.downloadsize
+    // @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_downloadsize(DnfPackage * pkg)
+    unsigned long long get_package_size() const;
+
+    /// @return Size the RPM package should occupy after installing on disk (RPMTAG_LONGSIZE).
+    ///         The information is always present - it is retrieved from rpmdb if the package is installed or from repodata if the package is available.
+    /// @since 5.0
```

Comment:
Really 5.0? What about 1.0.0. The current version is 0.65.0. In future we can increase the version.

Reply:
Good point. Let's discuss this next time we meet.
I used package version. Maybe it would be worth using API version independent on the package version.